include/libtorrent/span.hpp

```diff
@@ -105,12 +105,19 @@ namespace libtorrent
 			return { data() + offset, count };
 		}
 
-		T& operator[](std::size_t const idx)
+		T& operator[](std::size_t const idx) const
 		{
 			TORRENT_ASSERT(idx < m_len);
 			return m_ptr[idx];
 		}
 
+		T& operator[](int const idx) const
```

Comment:
yeah, I thought about that a bit. you mean that ``span`` should be ready to be replaced by gsl or some other standard type? so this utility feature may get in the way. I agree with that, they should be sepate, I just wasn't sure it was worth separating them up-front, rather then put it off until it's actually needed